src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,110 @@
-import React from 'react';
+import cn from 'classnames';
+import React, { useState } from 'react';
+import { Post } from '../types/Post';
+import { client } from '../utils/fetchClient';
+import { Comment } from '../types/Comment';
+
+interface Props {
+  selectedPost: Post | null;
+  setCommentsOfPost: (comments: Comment[] | null
+  | ((prevVar: Comment[] | null) => Comment[] | null)) => void;
```

Comment:
You can check what type setState function has and use types from react instead `React.Dispatch<React.SetStateAction<Comment[] | null>>`